File: WebContent/resources/js/paging.js

	//이전 버튼 이벤트
	function prev(type, page, range, rangeSize) {
			var page = ((range - 2) * rangeSize) + 1;
			var range = range - 1;
	
			var url = type;
			url = url + "?page=" + page;
			url = url + "&range=" + range;
	
			location.href = url;
		}
		
  	//페이지 번호 클릭
	function pagination(type, page, range, rangeSize, searchType, keyword) {
		var url = type;
		url = url + "?page=" + page;
		url = url + "&range=" + range;

		location.href = url;	

	}
	
	//다음 버튼 이벤트
	function next(type, page, range, rangeSize) {
		var page = parseInt((range * rangeSize)) + 1;
		var range = parseInt(range) + 1;

		var url = type;
		url = url + "?page=" + page;
		url = url + "&range=" + range;

		location.href = url;
	}

File: src/main/java/teamproject/sagi/dto/Paging.java
package teamproject.sagi.dto;

public class Paging {
	private int listSize = 15;	//한 페이지에 보일 게시글 갯수
	private int rangeSize = 10;	//목록갯수
	private int page;
	private int range;	//목록단위의 수
	private int listCnt;	//총 게시글 수
	private int pageCnt;
	private int startPage;
	private int startList;
	private int endPage;
	private boolean prev;
	private boolean next;
	public int getListSize() {
		return listSize;
	}
	public void setListSize(int listSize) {
		this.listSize = listSize;
	}
	public int getRangeSize() {
		return rangeSize;
	}
	public void setRangeSize(int rangeSize) {
		this.rangeSize = rangeSize;
	}
	public int getPage() {
		return page;
	}
	public void setPage(int page) {
		this.page = page;
	}
	public int getRange() {
		return range;
	}
	public void setRange(int range) {
		this.range = range;
	}
	public int getListCnt() {
		return listCnt;
	}
	public void setListCnt(int listCnt) {
		this.listCnt = listCnt;
	}
	public int getPageCnt() {
		return pageCnt;
	}
	public void setPageCnt(int pageCnt) {
		this.pageCnt = pageCnt;
	}
	public int getStartPage() {
		return startPage;
	}
	public void setStartPage(int startPage) {
		this.startPage = startPage;
	}
	public int getStartList() {
		return startList;
	}
	public void setStartList(int startList) {
		this.startList = startList;
	}
	public int getEndPage() {
		return endPage;
	}
	public void setEndPage(int endPage) {
		this.endPage = endPage;
	}
	public boolean isPrev() {
		return prev;
	}
	public void setPrev(boolean prev) {
		this.prev = prev;
	}
	public boolean isNext() {
		return next;
	}
	public void setNext(boolean next) {
		this.next = next;
	}
	
	public void pageInfo(int page, int range, int listCnt) {
		this.page = page;
		this.range = range;
		this.listCnt = listCnt;
		
		this.pageCnt = (int) Math.ceil(listCnt/(double)listSize);
		this.startPage = (range-1)*rangeSize+1;
		this.endPage = range*rangeSize;
		
		if(range == 1) {
			this.prev = false;
		}else {
			this.prev = true;
		}
		
		if(endPage >= pageCnt) {
			this.next = false;
			this.endPage = this.pageCnt;
		}else {
			this.next = true;
		}
	}
}
	
	